clang/lib/Driver/Driver.cpp

```diff
@@ -2465,8 +2484,9 @@ class OffloadingActionBuilder final {
           return ABRT_Success;
 
         // Replicate inputs for each GPU architecture.
-        auto Ty = IA->getType() == types::TY_HIP ? types::TY_HIP_DEVICE
-                                                 : types::TY_CUDA_DEVICE;
+        auto Ty = AssociatedOffloadKind == Action::OFK_HIP
```

Comment:
some of this looks like a spurious delta between trunk and amd-stg-open, perhaps we should propose a (mostly whitespace) patch to amd-stg-open based on trunk to get the signal/noise up